packages/@uppy/provider-views/src/SearchProviderView/Header.js

```diff
@@ -1,9 +1,32 @@
 const { h } = require('preact')
 
 module.exports = (props) => {
+  const { searchTerm, i18n, search } = props
+
+  const handleKeyPress = (ev) => {
+    if (ev.keyCode === 13) {
+      ev.stopPropagation()
+      ev.preventDefault()
+      search(ev.target.value)
+    }
+  }
+
   return (
-    <button type="button" onClick={props.triggerSearchInput} className="uppy-u-reset uppy-ProviderBrowser-userLogout">
-      {props.i18n('backToSearch')}
-    </button>
+    <div class="uppy-ProviderBrowser-search">
+      <input
+        class="uppy-u-reset uppy-ProviderBrowser-searchInput"
+        type="text"
+        placeholder={i18n('search')}
+        aria-label={i18n('search')}
+        value={searchTerm}
+        onKeyUp={handleKeyPress}
+        onKeyDown={handleKeyPress}
+        onKeyPress={handleKeyPress}
```

Comment:
Doesn't this fire way too many search calls? and therefor API requests? I think in general you want to at least debounce user input that does network requests, and only do one event instead of three.